src/fsharp/FSharp.DependencyManager/FSharp.DependencyManager.fs

```diff
@@ -0,0 +1,199 @@
+        match String.IsNullOrEmpty(s), String.IsNullOrEmpty(v) with
+        | true, true -> s + ";" + v
+        | true, false -> s
+        | false, true -> v
+        | _  -> ""
+
+    do
+#if !FX_NO_APP_DOMAINS
+        if AppDomain.CurrentDomain.ShadowCopyFiles then
+            raise (InvalidOperationException(sprintf "Nuget package manager can not work when shadow copy files enabled"))                      // @@@@@@@@@@@@@@@@@@@@@@@ Globalize me please
+#endif
+        emitFile (Path.Combine(scriptsPath, "Library.fs")) "namespace lib"
+
+#if !NETSTANDARD1_6
+    do if deleteAtExit then AppDomain.CurrentDomain.ProcessExit |> Event.add(fun _ -> deleteScripts )
+#endif
+
+    let formatPackageReference p =
+        let { Include=inc; Version=ver; RestoreSources=src } = p
+        seq {
+            match not (String.IsNullOrEmpty(inc)), not (String.IsNullOrEmpty(ver)) with
+            | true, true -> yield sprintf @"  <ItemGroup><PackageReference Include = '%s' Version = '%s' /></ItemGroup>" inc ver
+            | true, false -> yield sprintf @"  <ItemGroup><PackageReference Include = '%s' /></ItemGroup>" inc 
+            | _ -> ()
+            match not (String.IsNullOrEmpty(src)) with
+            | true -> yield sprintf @"  <PropertyGroup><RestoreSources>%s</RestoreSources></PropertyGroup>" (concat "$(RestoreSources)" src)
+            | _ -> ()
+        }
+
+    member __.Name:string = "MsBuild Nuget DependencyManager"
+
+    member __.Key:string = "nuget"
+
+    member __.ResolveDependencies(scriptDir:string, mainScriptName:string, scriptName:string, packageManagerTextLines:string seq) : bool * string list *  string list =
+
+        let packageReferences, loggerVerbosity =
+            let validatePackageName package packageName =
+                if String.Compare(packageName, package, StringComparison.OrdinalIgnoreCase) = 0 then
+                    raise (ArgumentException(sprintf "PackageManager can not reference the System Package '%s'" packageName))               // @@@@@@@@@@@@@@@@@@@@@@@ Globalize me please
+
+            let mutable loggerVerbosity = ""
+            let references = [
+                for line in packageManagerTextLines do
+                    let options = getOptions line
+                    let mutable found = false
+                    let mutable packageReference = { Include = ""; Version = ""; RestoreSources = "" }
+                    for opt in options do
+                        let addInclude v =
+                            // TODO:  Consider a comprehensive list of dotnet framework packages that are disallowed
+                            validatePackageName v "mscorlib"                                                    // Even though it's not a real package.
+                            validatePackageName v "FSharp.Core"
+                            validatePackageName v "System.ValueTuple"
+                            validatePackageName v "NETStandard.Library"
```

Comment:
We may want to consider `Microsoft.NETCore.App` and/or whatever gives us access to `Span`. I suppose that's a bit tricky, but Iactually just saw some requests about [how to use Span with FSI](https://stackoverflow.com/questions/53070089/how-to-use-span-in-f-interactive), and the only way to do that here would be to reference the relevant bits.